pkg/auth/oauth2authenticator.go

```diff
@@ -365,7 +372,9 @@ func (o *oAuth2Authenticator) authenticateWithAuthorizationCode() error {
 		return err
 	}
 
-	o.persistToken(token)
+	if err = o.persistToken(token); err != nil {
```

Comment:
Suggestion: How about returning err always. I think the code can be made easier here
```suggestion
	err = o.persistToken(token)
	return err
```